metaflow/plugins/kfp/kfp.py

```diff
@@ -74,17 +206,26 @@ def kfp_pipeline_from_flow():
         by invoking `step_container_op` for every step in the flow and handling the order of steps.
         """
 
-        # Store the list of steps in reverse order
-        step_container_ops = [step_container_op(step, code_url) for step in reversed(steps)]
-
-        # Each step in the list can only be executed after the next step in the list, i.e., list[-1] is executed first, followed
-        # by list[-2] and so on.
-        for i in range(len(steps) - 1):
-            step_container_ops[i].after(step_container_ops[i + 1])
+        step_container_ops = []
+        pre_start_op = (pre_start_container_op())(code_url)
+        step_container_ops.append(pre_start_op)
+
+        for step in steps:
+            prev_step_outputs = step_container_ops[-1].outputs
+            step_container_ops.append(
+                (step_container_op())(step, code_url,
+                      prev_step_outputs['ds_root'],
+                      prev_step_outputs['run_id'],
+                      prev_step_outputs['next_task_id'],
+                      prev_step_outputs['current_step'],
+                      prev_step_outputs['current_task_id']
```

Comment:
Can you just do `**prev_step_outputs` here instead?

Reply:
I've removed this code in the latest branch-and-join PR as the code no longer uses outputs from each step.